tools/c7n_azure/c7n_azure/session.py

```diff
@@ -13,21 +13,76 @@
 # limitations under the License.
 
 import importlib
-
+import os
+import logging
 from azure.cli.core.cloud import AZURE_PUBLIC_CLOUD
 from azure.cli.core._profile import Profile
+from azure.common.credentials import ServicePrincipalCredentials, BasicTokenAuthentication
 
 
 class Session(object):
 
     def __init__(self):
-        (self.credentials,
-         self.subscription_id,
-         self.tenant_id) = Profile().get_login_credentials(
-            resource=AZURE_PUBLIC_CLOUD.endpoints.active_directory_resource_id)
+        self.log = logging.getLogger('custodian.azure.session')
+        self.__provider_cache = {}
```

Comment:
we generally avoid the double underscore, vs single for private to avoid the mangling.